From 2c2cfadbb70a8d5d8957655ebebfb28a449c4fd2 Mon Sep 17 00:00:00 2001
From: Don Darling <don.osc2@gmail.com>
Date: Thu, 11 Mar 2010 13:43:37 -0600
Subject: [PATCH] Fix for playbin2.

playbin2 does not create audio/video sink elements when flags property is set
to configure the native audio/video sink.

See https://bugzilla.gnome.org/show_bug.cgi?id=606687 for more information.
---
 gst/playback/gstplaysink.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/gst/playback/gstplaysink.c b/gst/playback/gstplaysink.c
index fb39935..b09c316 100644
--- a/gst/playback/gstplaysink.c
+++ b/gst/playback/gstplaysink.c
@@ -1880,12 +1880,13 @@ gst_play_sink_reconfigure (GstPlaySink * playsink)
      * pick one and ignore the other then instead of erroring out?) */
     if (need_text && need_subp)
       goto subs_and_text;
-  } else if (flags & GST_PLAY_FLAG_VIDEO && playsink->video_pad) {
+  } else if (((flags & GST_PLAY_FLAG_VIDEO) || 
+    (flags & GST_PLAY_FLAG_NATIVE_VIDEO)) && playsink->video_pad) {
     /* we have video and we are requested to show it */
     need_video = TRUE;
   }
   if (playsink->audio_pad) {
-    if (flags & GST_PLAY_FLAG_AUDIO) {
+    if ((flags & GST_PLAY_FLAG_AUDIO) || (flags & GST_PLAY_FLAG_NATIVE_AUDIO)) {
       need_audio = TRUE;
     }
     if (playsink->audio_pad_raw) {
-- 
1.6.3.3

